MetaMask/metamask-mobile
# feat: js bundle size ci gate

## **Description**
CI Gate for JS Bundle Size
This PR will generate a js bundle as apart of the CI process and provide a configurable gate to fail the build if some size threshold is exceeded




## **Related issues**

https://github.com/MetaMask/metamask-mobile/issues/10671
https://consensyssoftware.atlassian.net/browse/INFRA-2045

## **Manual testing steps**

1. Example setting max js bundle size to 25MB with expected bundle size of ~31MB resulted in a failed CI run https://github.com/MetaMask/metamask-mobile/actions/runs/10604838423/job/29392430371?pr=10876
2. Example setting max js bundle size to 50MB with expected bundle size of ~31MB resulted in a successful CI run https://github.com/MetaMask/metamask-mobile/actions/runs/10604977707/job/29392876282?pr=10876

## **Screenshots/Recordings**


N/A - no mobile changes, strictly CICD

### **Before**



### **After**



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
